click_option_group/_core.py

```diff
@@ -351,3 +351,26 @@ def handle_parse_result(self, option: GroupedOption, ctx: click.Context, opts: d
                           f'"{self.get_default_name(ctx)}" option group:')
             error_text += f'\n{self.get_error_hint(ctx)}'
             raise click.UsageError(error_text, ctx=ctx)
+
+
+class AllOptionGroup(OptionGroup):
+    """Option group with required all/none options of this group
+
+    `AllOptionGroup` defines the behavior:
+    - All options from the group must be set or None must be set.
+    """
+
+    @property
```

Comment:
I think we need to add `required` and `hidden` option attributes to the list of forbidden attributes for this class:
https://github.com/click-contrib/click-option-group/blob/a47e2216bc5c006824ddc0602662b2db03c714bd/click_option_group/_core.py#L286-L288

Reply:
@espdev, Updated